seastar/src/cxx_async_futures.hh

```diff
@@ -0,0 +1,7 @@
+#include "rust/cxx.h"
+#include "cxx-async/include/rust/cxx_async.h"
+#include "cxx-async/include/rust/cxx_async_seastar.h"
+#include <seastar/core/coroutine.hh>
+
+CXXASYNC_DEFINE_FUTURE(void, seastar_ffi, config_and_start_seastar, VoidFuture);
```

Comment:
I think we'd rather have any `cxx-async` futures to be declared directly under the `seastar_ffi` namespace, as they won't be unique to the context of `config_and_start_seastar.*` but will be used in other places as well (`seastar_ffi::timer` and `seastar_ffi::spawn` to name a few). Of course they can still be used there, but the second namespace is slightly misleading.

Reply:
I think that the easiest way to achieve this is by changing
```rust
CXXASYNC_DEFINE_FUTURE(void, seastar_ffi, config_and_start_seastar, VoidFuture);
```
to
```rust
CXXASYNC_DEFINE_FUTURE(void, seastar_ffi, VoidFuture);
```
and by defining `VoidFuture` in `extern "C++"` block like that
```rust
#[namespace = "seastar_ffi"]
type VoidFuture = crate::cxx_async_futures::VoidFuture;
```
in every `ffi` module that uses `VoidFuture`. At least this what I've done to be able to use `seastar_ffi::<some_other_namespace>` instead of `seastar_ffi::config_and_start_seastar` in another task.

Reply:
Yes, I agree that this is a good idea. Changed.